### PR TITLE
Test

### PR DESCRIPTION
**Pull request by mxxt1**
# Description

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/components/Profile/Profile.jsx [hunk 1]
```diff
@@ -26,22 +26,22 @@ const Profile = ({ apiKey, setApiKey }) => {
...
+      
       const response = await axios.post(
         '/api/apikeyRoute/private',
         {
           id: sub
         },
         {
           headers: { Authorization: `Bearer ${token}` },
-          baseURL: 'https://sauti-africa-market-master.herokuapp.com/'
+          baseURL: 'https://sauti-africa-market-staging-3.herokuapp.com/'
```
Comment (blackmacaroon) [inline src/components/Profile/Profile.jsx]: switch to master?

File: src/components/Profile/UserRoles.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { useAuth0 } from '../../contexts'
+import axios from 'axios'
+import './UserRoles.scss'
+
+const UserRoles = () => {
+  const { user } = useAuth0()
+
+  const updateUserRole = (userId, user) => {
+    axios
+      .put(`https://sauti-africa-market-staging-3.herokuapp.com/api/users/${userId}`, user)
```
Comment (blackmacaroon) [inline src/components/Profile/UserRoles.jsx]: switch to master?

File: src/utils/roles.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// ! HELPER FUNCTIONALITIES FOR DATA PERTAINING TO USER ROLES
+
+// * PKGS
+import axios from 'axios';
+
+// * RETURN THE ROLE OF THE LOGGED IN USER AND SET STATE DATA
+export const validateRole = async (data, set) => axios.post(
+    'http://localhost:8888/api/users/',
```
Comment (blackmacaroon) [inline src/utils/roles.js]: should this still be local host?

File: now.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "build": {
+        "env": {
+            "REACT_APP_GOOGLE_TRACKING_ID": "@UA-150653982-1",
+            "REACT_APP_AUDIENCE": "@https://sauti-africa-market-prices.auth0.com/api/v2/",
+            "REACT_APP_DOMAIN": "@sauti-africa-market-prices.auth0.com",
+            "REACT_APP_CLIENT_ID": "@bxosYDeCRsMMIEuRFyte2kxifC5uK9ua",
+            "REACT_APP_HOST": "@http://localhost:3000"
```
Comment (blackmacaroon) [inline now.json]: should this still be local host?

File: src/hooks/useAuth0.jsx [hunk 5]
```diff
@@ -67,6 +83,25 @@ export const Auth0Provider = ({
     setUser(user)
   }
 
+  // * GET AND SET USER ROLE IF AUTHENTICATED
+  const getRole = () => axios
+    // .post(`https://sauti-africa-market-staging-3.herokuapp.com/api/users/`, user)
+    .post(`http://localhost:8888/api/users/`, user)
```
Comment (blackmacaroon) [inline src/hooks/useAuth0.jsx]: should this still be local host?

File: cypress/integration/userRoles.spec.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  sub: 'google-oauth2|111661795839592168840'
+}
+
+const role = { app_metadata: {
+  role: "freeUser"
+}}
+
+describe('Request', () => {
+  it('Gets user data from the backend', () => {
+    cy.request('POST', 'http://localhost:8888/api/users', userId).should(
```
Comment (blackmacaroon) [inline cypress/integration/userRoles.spec.js]: even for testing, this should be the staging db maybe?

File: cypress/integration/userRoles.spec.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        expect(response).to.have.property('duration')
+      }
+    )
+  })
+})
+
+
+describe('Request', () => {
+  it('Gets user data from the backend', () => {
+    cy.request('PUT', `http://localhost:8888/api/users/google-oauth2|111661795839592168840`, role).should(
```
Comment (blackmacaroon) [inline cypress/integration/userRoles.spec.js]: even for testing, this should be the staging db maybe?

File: src/utils/roles.js (hunk 3 above)
Comment (mxxt1) [reply src/utils/roles.js]: Need clarity from Bao on this, will resolve asap.

File: now.json (hunk 4 above)
Comment (mxxt1) [reply now.json]: I don't think we need this file anymore, I believe it was the config file for zeit

File: src/hooks/useAuth0.jsx (hunk 5 above)
Comment (mxxt1) [reply src/hooks/useAuth0.jsx]: Nope!

**Pull request merged**